snow/engine/avalanche/transitive_test.go

```diff
@@ -4947,3 +4947,223 @@ func TestEngineTransitionDependencyAbandoned(t *testing.T) {
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Accepted,
+	}}
+	trA := &conflicts.TestTransition{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Unknown,
+	}
+	trB := &conflicts.TestTransition{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Unknown,
+	}
+	trC := &conflicts.TestTransition{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Unknown,
+		DependenciesV: []conflicts.Transition{
+			trA,
+			trB,
+		},
+	}
+	txA1 := &conflicts.TestTx{
+		BytesV: []byte{0},
+		EpochV: priorEpoch,
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		TransitionV: trA,
+	}
+	txA2 := &conflicts.TestTx{
+		BytesV: []byte{1},
+		EpochV: currentEpoch,
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		TransitionV: trA,
+	}
+	txB := &conflicts.TestTx{
+		BytesV: []byte{2},
+		EpochV: priorEpoch,
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		TransitionV: trB,
+	}
+	txC := &conflicts.TestTx{
+		BytesV: []byte{2},
+		EpochV: priorEpoch,
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		TransitionV: trC,
+	}
+	vtxA1 := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		EpochV:   priorEpoch,
+		ParentsV: []avalanche.Vertex{gVtx},
+		HeightV:  1,
+		TxsV:     []conflicts.Tx{txA1},
+		BytesV:   utils.RandomBytes(32),
+	}
+	vtxA2 := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		EpochV:   currentEpoch,
+		ParentsV: []avalanche.Vertex{gVtx},
+		HeightV:  1,
+		TxsV:     []conflicts.Tx{txA2},
+		BytesV:   utils.RandomBytes(32),
+	}
+	vtxB := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		EpochV:   priorEpoch,
+		ParentsV: []avalanche.Vertex{gVtx},
+		HeightV:  1,
+		TxsV:     []conflicts.Tx{txB},
+		BytesV:   utils.RandomBytes(32),
+	}
+	vtxC := &avalanche.TestVertex{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Unknown,
+		},
+		EpochV:   priorEpoch,
+		ParentsV: []avalanche.Vertex{vtxA1, vtxB},
+		HeightV:  2,
+		TxsV:     []conflicts.Tx{txC},
+		BytesV:   utils.RandomBytes(32),
+	}
+
+	// Tell engine to issue vtxA2 by sending push query from [vdr]
+	manager.ParseF = func(b []byte) (avalanche.Vertex, error) {
+		assert.Equal(t, b, vtxA2.Bytes())
+		vtxA2.StatusV = choices.Processing
+		txA2.StatusV = choices.Processing
+		trA.StatusV = choices.Processing
+		return vtxA2, nil
+	}
+	manager.GetF = func(id ids.ID) (avalanche.Vertex, error) {
+		if id != vtxA2.ID() {
+			t.Fatalf("Called Get for unexpected vtxID: %s", id)
```

Comment:
I think `assert.FailNow` is nicer because it gives a stacktrace but not a big deal